clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -358,6 +452,15 @@ public int sessionId() {
                 .setResponses(topicResponseList);
     }
 
+    private Boolean supportsTopicIds() {
+        return data.responses().stream().findFirst().filter(
+            topic -> !topic.topicId().equals(Uuid.ZERO_UUID)).isPresent();
```

Comment:
Does one non-zero id mean we have all ids?

Reply:
The idea is that we should only be able to send this request version if we had an id for each topic. I do need to take a closer look at this